src/lib.rs

```diff
@@ -906,13 +908,31 @@ impl Pandoc {
         // apply all filters
         let filtered = filters.into_iter().fold(o, |acc, item| item(acc));
         self.input = Some(InputKind::Pipe(filtered));
+        Ok(())
+    }
+    
+    /// actually execute pandoc
+    pub fn execute(mut self) -> Result<(), PandocError> {
+        let _ = try!(self.preprocess());
         let _ = try!(self.run());
         Ok(())
     }
+
+    /// Execute pandoc and return the result from stdout.
+    pub fn execute_with_output(mut self) -> Result<String, PandocError> {
```

Comment:
Yes, let's plan to do that; I'll submit a follow-up commit sometime in the next week or so.